content/en/docs/concepts/cluster-administration/system-logs.md

```diff
@@ -0,0 +1,113 @@
+```
+
+[command-line-tools-reference]: /docs/reference/command-line-tools-reference
+[klog]: https://github.com/kubernetes/klog
+
+### Structured Logging (ALPHA)
+
+Structured logging is a effort to introduce a uniform structure in log messages allowing for easy extraction of information, making logs easier and cheaper to store and process.
+New message format is backward compatible and enabled by default.
+
+Format of structured logs:
+```
+<klog header> "<message>" <key1>="<value1>" <key2>="<value2>" ...
+```
+
+Example:
+```
+I1025 00:15:15.525108       1 controller_utils.go:116] "Pod status updated" pod="kube-system/kubedns" status="ready"
+```
+
+**WARNING** AS structured format is an ALPHA feature, details like exact format or value serialization are a subject to change.
+
+**WARNING** Currently only a small portion of logs utilize structured message format, migration to currently ongoing and is expect to take very long time before every log call is migrated. If you intend to parse logs, please make sure to handle normal messages.
+
+### JSON log format (ALPHA)
+
+By providing `--logging-format=json` flag to commandline we can change format of logs written by Kubernetes System components.
+JSON format allows for much more efficient log generation and is more concise format.
+
+Example of JSON log format (pretty printed):
+```json
+{
+   "ts": 1580306777.04728,
+   "v": 4,
+   "msg": "Pod status updated",
+   "pod":{
+      "name": "nginx-1",
+      "namespace": "default"
+   },
+   "status": "ready"
+}
+```
+**WARNING** Not all logs are guaranteed to be written in JSON format (e.g. during process start). If you intend to parse logs, please make sure to can handle non JSON logs.
+
+Keys with special meaning:
+* `ts` - timestamp as Unix time (required, float)
+* `v` - verbosity (required, int, default 0)
+* `err` - error string (optional, string)
+* `msg` - message (required, string)
+
+**WARNING** AS JSON format is an ALPHA feature, details like name of fields or JSON serialization are a subject to change.
+
+List of components currently supporting JSON format:
+* {{< glossary_tooltip term_id="kube-controller-manager" text="kube-controller-manager" >}}
+* {{< glossary_tooltip term_id="kube-apiserver" text="kube-apiserver" >}}
+* {{< glossary_tooltip term_id="kube-scheduler" text="kube-scheduler" >}}
+* {{< glossary_tooltip term_id="kubelet" text="kubelet" >}}
+
+**WARNING** JSON format doesn't support most preexisting klog flags. Component will exit if provided with both `--logging-format=json` and any unsupported `klog` flag. Please read [command-line-tools-reference] to find what flags are supported.
+
+### Log verbosity level
+
+By controlling verbosity (using `-v` command line flag) we can filter logs based on how severe they are, controlling resulting log volume.
+The higher verbosity of log the less severe they are, with verbosity 0 being used for logs that should ALWAYS be visible.
+The practical default level is 2.
```

Comment:
1) What is the full range of numbers you can use with `-v`?  (it starts at 0, 2 is "practical" for daily use... but how high can I go?)
2) How do I use `-v` – i.e. `-v=0`?
3) Is there any way we can describe what gets added to the logs with each number increment? i.e.

- 0 = essential messages, like system errors
- 1 = 
- 2 = lists successful events (or something)
- 3 = 
- 4 = 

etc.?

if this is impractical feel free to ignore it.

Reply:
Ad 1, don't think there is a limit, guidelines give usecase up to 5, highest that make sense was 10 (there is 12, but looks like mistake). 
Ad 2 Yes, 
Ad 3 We have [guide for developers](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/logging.md) (linked in what's next). But I would not suggest to detail levels in documentation as I don't think they strictly complied to, so we shouldn't promise it to users.

Reply:
It would help a lot to specify that logging verbosity is an integer, and to provide a range.

Reply:
@serathius: I would take the list from the doc you just linked and include it in this documentation :). This is valuable information!

Reply:
This doc is old developers guide, some parts are outdated. As this is guidelines, I don't expect them to be very accurate. I would be worried about adding this to documentation.

Verbosity level values are a evolving and can we cannot give any guarantees about where what type of log is where (in current state). Creating a guide was meant to give contributors some intuition about levels, but not hard rules. 

For example, this guide lists HTTP logs on V(2), but I'm pretty sure that they are on V(3) currently. Copy&Pasting this guide will result in bad experience of users, when they will try to base their configuration on incorrect information in guide. 

Overall I think we should convey that users should experiment with levels and select one based on costs they are willing to pay for log volume generated.

Reply:
Okay then. For now let's go with that approach :)